### ISSUE TITLE
Doesn't seem to work on iOS

### ISSUE THREAD

**Issue by kmiller15211** [mustafa-707/unlock_detector#1]
Here is my sample code that's inside of my main.dart: 

Object instance: 
```
final unlockDetector = UnlockDetector();
```

And my use of said object, which is at the end of my test application's main(): 
```
    unlockDetector.stream?.listen((event) {
      mainLogger.info("************ KCM unlock status: $event *************");
    });
```

On my Pixel 7 Android, I get the following: 
```
D/unlock_detector(27974): Screen locked
I/flutter (27974): [2024/07/26] [15:00:20.139] INFO - Main: ************ KCM unlock status: UnlockDetectorStatus.locked *************
...
I/flutter (27974): [2024/07/26] [15:00:21.226] INFO - Main: ************ KCM unlock status: UnlockDetectorStatus.screenOn *************
I/flutter (27974): [2024/07/26] [15:00:21.229] INFO - Main: ************ KCM unlock status: UnlockDetectorStatus.unlocked *************
```

But, on iOS (iPhone 13, iOS Version 17.5.1), I don't get the lock/unlock messages at all. I have notifications through AppLifecycleState, and I see those come through, but I do not see the lock/unlock: 
```
flutter: [2024/07/26] [15:12:50.560] INFO - UI: AppLifecycleState.hidden
flutter: [2024/07/26] [15:12:50.561] INFO - UI: AppLifecycleState.inactive
flutter: [2024/07/26] [15:12:50.841] INFO - UI: AppLifecycleState.resumed
flutter: [2024/07/26] [15:13:41.550] INFO - UI: AppLifecycleState.inactive
...
flutter: [2024/07/26] [15:13:42.551] INFO - UI: AppLifecycleState.hidden
flutter: [2024/07/26] [15:13:42.552] INFO - UI: AppLifecycleState.paused
flutter: [2024/07/26] [15:13:42.592] INFO - UI: Gateway Screen
flutter: [2024/07/26] [15:13:42.594] INFO - UI: Home Page
```

**Comment by mustafa-707** [comment]
@kmiller15211  Thank you for reporting , i just did a fix , can you confirm that version `v0.1.0` is working ?

**Comment by kmiller15211** [comment]
Yep, that seemed to fix it, thank you! Now on iOS: 

=================================
```
flutter: [2024/07/29] [08:07:05.374] INFO - UI: AppLifecycleState.hidden
flutter: [2024/07/29] [08:07:05.375] INFO - UI: AppLifecycleState.paused
flutter: [2024/07/29] [08:07:05.375] INFO - Main: ************ KCM unlock status: UnlockDetectorStatus.locked *************
flutter: [2024/07/29] [08:07:08.207] INFO - UI: AppLifecycleState.hidden
flutter: [2024/07/29] [08:07:08.208] INFO - UI: AppLifecycleState.inactive
flutter: [2024/07/29] [08:07:08.210] INFO - Main: ************ KCM unlock status: UnlockDetectorStatus.unlocked *************
flutter: [2024/07/29] [08:07:08.819] INFO - UI: AppLifecycleState.resumed
```
